connector_office_365_hr_leave/models/hr_leave.py

```diff
@@ -50,3 +51,41 @@ def _validate_leave_request(self):
             if not leave.meeting_id and leave.meeting_id.user_id:
                 continue
             leave._office_365_push()
+
+    @api.multi
+    def action_refuse(self):
```

Comment:
We need to override a lot of things which is a pity...
Wouldn't this work as anyway the user we pass is always the user of the related meeting?

```
class HolidaysRequest(models.Model):

    @api.multi
    def action_refuse(self):
        return super(HolidaysRequest, self.with_context(o365_unlink_as_event_user=True).action_refuse()
    
class CalendarEvent(models.Model):

    @api.multi
    def unlink(self):
        if self.env.context.get('o365_unlink_as_event_user'):
            # loop on leaves, call super with event.user_id.id in 'user' context
```

Or even modify connector_office_365/models/calendar_event.py to check `o365_unlink_as_event_user` and set `user` as `event.user_id.id`

Reply:
yes this way it would work. i'll change it.